### ISSUE TITLE
New Rule: space-after-function [$15 awarded]

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1276]
Similar to `space-after-keywords` this rule should ensure consistent spacing after the function keyword.

Valid:

``` js
function foo() {};

function () {};
```

Invalid:

``` js
function() {}
```
##
